### DISCUSSION TITLE
Email or Phone Verification in the Case of Lost 2FA Access

### DISCUSSION THREAD

**Discussion by pinchese** [stoatchat/discussions#428] [Feature Suggestions]
hi, i recently messed up my 2fa app, and while i was able to recover most of them by way of removing 2fa from the accounts entirely, which happened by either getting an email with a one time code, or an sms message also with a one time code. i've also seen other apps let me use email verification for viewing backup codes, which may help

i'm requesting that you add this method of 2fa recovery, as i am currently unable to log into any new device because of the locked 2fa app, and if i log out of the only device i'm still logged in to, i will have completely lost my revolt account

anyway, thanks for this awesome app, yall did a great job with it

**Comment by sussycatgirl** [comment]
I don't think allowing users to disable 2FA via email, the same verification method used for resetting passwords, is a good idea from a security standpoint, and SMS is inherently insecure and will definitely not be used for 2FA.

The recovery codes you get while setting up 2FA are intended to be used for account recovery instead. If you don't have those, you might have luck getting into contact with support.

@insertish You might want to help out here

**Comment by pinchese** [comment]
i woke up today to a revolt app that signed me out, so yeah i'll just email support
thank you!
